presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -533,4 +565,37 @@ public Boolean process(Node node, @Nullable Void context)
             return super.process(node, context);
         }
     }
+
+    /**
+     * Verifies that no query node output columns are referenced from ORDER BY expressions.
+     * Only FROM columns are allowed within ORDER BY expressions.
+     */
+    private class VerifyOrderByAggregationVisitor
+            extends DefaultTraversalVisitor<Void, Void>
+    {
+        @Override
+        protected Void visitIdentifier(Identifier node, Void context)
+        {
+            verifyField(node, QualifiedName.of(node.getName()));
+            return null;
+        }
+
+        @Override
+        protected Void visitDereferenceExpression(DereferenceExpression node, Void context)
+        {
+            if (analysis.getColumnReferences().contains(node)) {
+                verifyField(node, DereferenceExpression.getQualifiedName(node));
+            }
+
+            return null;
+        }
+
+        private void verifyField(Expression node, QualifiedName qualifiedName)
+        {
+            ResolvedField resolvedField = orderByScope.get().resolveField(node, qualifiedName);
```

Comment:
This is OK as long as there is no nested scope, i.e. no subquery. This will be rightlyrejected by analysis 
```SELECT a AS b FROM t1 GROUP BY t1.a ORDER BY MAX((SELECT b))```
but this will be incorrectly rejected with `Column 'x' cannot be resolved`:
```SELECT a FROM t1 GROUP BY a ORDER BY MAX((SELECT x FROM (VALUES 4) t(x)))```

I think you need

```@Override
        protected Void visitSubqueryExpression(SubqueryExpression node, Void context)
        {
            return null;
        }```
here + TODO for myself to straighten this up.